neo/api/JSONRPC/test_json_rpc_api.py

```diff
@@ -370,72 +348,66 @@ def test_get_contract_state(self):
     def test_get_contract_state_0x(self):
         contract_hash = "0xb9fbcff6e50fd381160b822207231233dd3c56c2"
         req = self._gen_post_rpc_req("getcontractstate", params=[contract_hash])
-        mock_req = mock_post_request(json.dumps(req).encode("utf-8"))
-        res = json.loads(self.app.home(mock_req))
+        res = json.loads(self.do_test_post("/", json=req))
         self.assertEqual(res['result']['code_version'], '')
 
     def test_get_contract_state_not_found(self):
         contract_hash = '0xb9fbcff6e50fd381160b822207231233dd3c56c1'
         req = self._gen_post_rpc_req("getcontractstate", params=[contract_hash])
-        mock_req = mock_post_request(json.dumps(req).encode("utf-8"))
-        res = json.loads(self.app.home(mock_req))
+        res = json.loads(self.do_test_post("/", json=req))
         self.assertTrue('error' in res)
         self.assertEqual(res['error']['message'], 'Unknown contract')
 
     def test_get_raw_mempool(self):
-        # TODO: currently returns empty list. test with list would be great
+        nodemgr = self.api_server.nodemgr
+        nodemgr.reset_for_test()
+
+        raw_tx = b'd100644011111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111081234567890abcdef0415cd5b0769cc4ee2f1c9f4e0782756dabf246d0a4fe60a035400000000'
+        tx = Helper.AsSerializableWithType(binascii.unhexlify(raw_tx), 'neo.Core.TX.InvocationTransaction.InvocationTransaction')
+        nodemgr.mempool.add_transaction(tx)
+
         req = self._gen_post_rpc_req("getrawmempool", params=[])
-        mock_req = mock_post_request(json.dumps(req).encode("utf-8"))
-        res = json.loads(self.app.home(mock_req))
+        res = json.loads(self.do_test_post("/", json=req))
+
         mempool = res['result']
 
-        # when running only these tests, mempool is empty. when running all tests, there are a
-        # number of entries
-        if len(mempool) > 0:
-            for entry in mempool:
-                self.assertEqual(entry[0:2], "0x")
-                self.assertEqual(len(entry), 66)
+        self.assertEqual(1, len(mempool))
+        self.assertEqual(tx.Hash.To0xString(), mempool[0])
+        nodemgr.reset_for_test()
 
     def test_get_version(self):
-        # TODO: what's the nonce? on testnet live server response it's always 771199013
```

Comment:
Should was assert the nonce value in the test?

Reply:
added